WordPress/gutenberg
# Gallery Block Refactor: Add last target drop indicator

## Description
* Provides a drop indicator (blue vertical line) for the last position in a block list when the appender is being hidden via `renderAppender: false`
* Prevents border collapsing while gallery images are being dragged and dropped.
    * This may or may not be desired as number of images per row can vary from the gallery's column count while dragging

The last target drop indicator has only been added at this stage for horizontal orientations and when the appender is being hidden. It could be expanded if this approach was sound and there was an actual need for it.

**Note: There is a current bug where reordering of gallery images throws a "callback is not a function" error in console. This can cause some image selection issues after dragging and dropping the first time**

## How has this been tested?
Manually.

1. Checkout this PR, create a post and add a gallery with multiple images
2. Drag a single image into a new position, confirming gallery image order updates correctly
3. Drag a single image to the last position with the gallery, there should be a blue drop indicator shown
4. Repeat steps 2 & 3 while dragging multiple image blocks

## Screenshots

| Before | After | After - Multiple Images |
|-------|-------|------------------------|
| ![DropIndicator](https://user-images.githubusercontent.com/60436221/104433617-07f82f80-55d6-11eb-92d9-a50b733803bd.gif) | ![DropIndicator2](https://user-images.githubusercontent.com/60436221/104433633-0c244d00-55d6-11eb-917a-c9c4e949d7c3.gif) | ![DropIndicator3](https://user-images.githubusercontent.com/60436221/104433756-2eb66600-55d6-11eb-8629-9fe5337f46c2.gif) |

## Types of changes
Enhancement

## Next Steps
* Revert the margin removals while dragging if not satisfactory

## Checklist:
